src/palette/internal/palettecelliconengine.cpp

```diff
@@ -70,26 +70,24 @@ void PaletteCellIconEngine::paintCell(Painter& painter, const RectF& rect, bool
         return;
     }
 
-    painter.setPen(configuration()->elementsColor());
+    notation::IEngravingItemIconEngine::PaintParams params;
 
-    if (element->isActionIcon()) {
-        paintActionIcon(painter, rect, element, dpi);
-        return; // never draw staff for icon elements
-    }
+    const QColor color = configuration()->elementsColor();
+    params.color = &color;
 
-    const bool drawStaff = m_cell->drawStaff;
-    const qreal spatium = configuration()->paletteSpatium() * m_extraMag * m_cell->mag;
+    params.mag = m_extraMag * m_cell->mag;
+    params.xoffset = m_cell->xoffset;
+    params.yoffset = m_cell->yoffset;
 
-    PointF origin = rect.center(); // draw element at center of cell by default
-    if (drawStaff) {
-        const qreal topLinePos = paintStaff(painter, rect, spatium); // draw dummy staff lines onto rect.
-        origin.setY(topLinePos); // vertical position relative to staff instead of cell center.
-    }
+    params.rect = &rect;
+    params.dpi = &dpi;
+
+    const qreal spatium = configuration()->paletteSpatium() * params.mag;
+    params.spatium = &spatium;
 
-    painter.translate(origin);
-    painter.translate(m_cell->xoffset * spatium, m_cell->yoffset * spatium); // additional offset for element onlym
+    params.drawStaff = m_cell->drawStaff;
 
-    paintScoreElement(painter, element, spatium, drawStaff, dpi);
+    engravingIconEngine()->paintScoreElement(painter, element, params);
 }
 
 void PaletteCellIconEngine::paintBackground(Painter& painter, const RectF& rect, bool selected, bool current) const
```

Comment:
Shouldn't it also be moved to engravingIconEngine?

Reply:
I don't think so - this is the method that paints the cell background.